include/simdjson/inline/document.h

```diff
@@ -805,53 +808,38 @@ inline document::element_result document::object::operator[](const char *json_po
   return (*this)[std::string_view(json_pointer)];
 }
 inline document::element_result document::object::at(std::string_view json_pointer) const noexcept {
-  // Unescape the key
-  std::string unescaped;
-  unescaped.reserve(json_pointer.length());
-  size_t i;
-  for (i = 0; i < json_pointer.length() && json_pointer[i] != '/'; i++) {
-    switch (json_pointer[i]) {
-      // Handle ~ escaping: ~0 = ~, ~1 = /
-      case '~': {
-        i++;
-        // ~ at end of string is invalid
-        if (i >= json_pointer.length()) { RETURN_ERROR(INVALID_JSON_POINTER, "~ at end of string in JSON pointer"); }
-        switch (json_pointer[i]) {
-          case '0':
-            unescaped.push_back('~');
-            break;
-          case '1':
-            unescaped.push_back('/');
-            break;
-          default:
-            RETURN_ERROR(INVALID_JSON_POINTER, "Unexpected ~ escape character in JSON pointer");
-        }
-        break;
-      }
-      // TODO backslash doesn't appear to be a thing in JSON pointer
-      case '\\': {
-        i++;
-        // backslash at end of string is invalid
-        if (i >= json_pointer.length()) { RETURN_ERROR(INVALID_JSON_POINTER, "~ at end of string in JSON pointer"); }
-        // Check for invalid escape characters
-        if (json_pointer[i] != '\\' && json_pointer[i] != '"' && json_pointer[i] > 0x1F) {
-          RETURN_ERROR(INVALID_JSON_POINTER, "Invalid backslash escape in JSON pointer");
-        }
-        unescaped.push_back(json_pointer[i]);
-        break;
-      }
-      default:
-        unescaped.push_back(json_pointer[i]);
-        break;
-    }
-  }
+  size_t slash = json_pointer.find('/');
+  std::string_view key = json_pointer.substr(0, slash);
```

Comment:
I checked that substr does not create a copy of the underlying data, and indeed, it appears to be reasonable as per the specification... https://en.cppreference.com/w/cpp/string/basic_string_view/substr

Note that substr applied to std::string creates a whole new string.